pkg/reconciler/autoscaling/kpa/kpa_test.go

```diff
@@ -1156,22 +1158,6 @@ func TestControllerSynchronizesCreatesAndDeletes(t *testing.T) {
 	if !newKPA.Status.IsReady() {
 		t.Error("Status.IsReady() was false")
 	}
-
-	fakeservingclient.Get(ctx).ServingV1alpha1().Revisions(testNamespace).Delete(testRevision, nil)
-	fakerevisioninformer.Get(ctx).Informer().GetIndexer().Delete(rev)
-	fakeservingclient.Get(ctx).AutoscalingV1alpha1().PodAutoscalers(testNamespace).Delete(testRevision, nil)
-	fakepainformer.Get(ctx).Informer().GetIndexer().Delete(kpa)
-	if err := ctl.Reconciler.Reconcile(context.Background(), testNamespace+"/"+testRevision); err != nil {
-		t.Errorf("Reconcile() = %v", err)
-	}
-
-	if fakeDeciders.deleteCallCount.Load() == 0 {
-		t.Fatal("Decider was not deleted")
-	}
-
-	if fakeDeciders.deleteBeforeCreate.Load() {
-		t.Fatal("Deciders.Delete ran before OnPresent")
-	}
```

Comment:
Are we losing coverage here?

Reply:
Yes, but I lack confidence in our ability to test the informer hooks.  Honestly, I can't think of any good examples we have.  If you have one, please share (and "shut up and take my money").

Reply:
Well there's no good answer I fear. You'd have to run a "proper" controller and poll each of the states. I did fix some stuff recently, you might take a look at https://github.com/knative/serving/blob/318466bb9f2679945dfbd72647cbea73945d4037/pkg/reconciler/nscert/nscert_test.go#L311-L358 for example. These have been the most notorious wrt. flakes though (these == tests that start a proper controller with a workQueue and all).

Reply:
I think the most interesting aspect of the test coverage is likely what your example isn't testing though, which is the invocation of the change function: https://github.com/knative/serving/blob/318466bb9f2679945dfbd72647cbea73945d4037/pkg/reconciler/nscert/nscert_test.go#L345.  Given that we're using a fake decider, it's basically like calling delete ourselves (and we'd have to name the method to test things this way, I think).

LMK if I'm misunderstanding the reference.  +1 to these tests being flake central 😬

Reply:
Ehr, you're right the tests are setup through an informer themselves. So this is a whole indeed. Using a mocked `fakeDeciders` + a "proper" controller you should be able to poll the amount of times `Delete` is called though?

Yes, that sounds horrible.

Reply:
alright, this is back.